packages/server/src/integrations/rest.ts

```diff
@@ -481,59 +481,47 @@ export class RestIntegration implements IntegrationBase {
             }
           }
         )
-        const nodeForm = form as FormData & {
-          getHeaders?: () => Record<string, string | string[]>
-        }
-        const formHeaders = nodeForm.getHeaders?.()
-        if (formHeaders) {
-          const ensureHeaderObject = (): Record<
-            string,
-            string | readonly string[]
-          > => {
-            if (!input.headers) {
-              const headerObject: Record<string, string> = {}
-              input.headers = headerObject
-              return headerObject
-            }
-            if (Array.isArray(input.headers)) {
-              const headerObject = input.headers.reduce<Record<string, string>>(
-                (acc, [name, value]) => {
-                  acc[name] = value
-                  return acc
-                },
-                {}
-              )
-              input.headers = headerObject
-              return headerObject
-            }
-            if (input.headers instanceof Headers) {
-              const headerObject: Record<string, string> = {}
-              input.headers.forEach((value, key) => {
-                headerObject[key] = value
-              })
-              input.headers = headerObject
-              return headerObject
-            }
-            return input.headers
+        const ensureHeaderObject = (): Record<
+          string,
+          string | readonly string[]
+        > => {
+          if (!input.headers) {
+            const headerObject: Record<string, string> = {}
+            input.headers = headerObject
+            return headerObject
           }
-          const headers = ensureHeaderObject()
-          for (let [headerName, headerValue] of Object.entries(formHeaders)) {
-            const normalizedName = headerName.toLowerCase()
-            const existingKey = Object.keys(headers).find(
-              key => key.toLowerCase() === normalizedName
+          if (Array.isArray(input.headers)) {
+            const headerObject = input.headers.reduce<Record<string, string>>(
+              (acc, [name, value]) => {
+                acc[name] = value
+                return acc
+              },
+              {}
             )
-            if (existingKey) {
-              continue
-            }
-            if (typeof headerValue === "undefined" || headerValue === null) {
-              continue
-            }
-            const headerString = Array.isArray(headerValue)
-              ? headerValue.join(", ")
-              : String(headerValue)
-            headers[headerName] = headerString
+            input.headers = headerObject
+            return headerObject
           }
+          if (input.headers instanceof Headers) {
+            const headerObject: Record<string, string> = {}
+            input.headers.forEach((value, key) => {
+              headerObject[key] = value
+            })
+            input.headers = headerObject
```

Comment:
why mutate *and* return something else?

Reply:
Good point, now only returning the value and not mutating the input